numpy/core/include/numpy/npy_math.h

```diff
@@ -129,6 +129,91 @@ NPY_INLINE static float __npy_nzerof(void)
 #define NPY_COEFF_Q1_EXPf -2.742335390411667452936e-01f
 #define NPY_COEFF_Q2_EXPf 2.159509375685829852307e-02f
 
+/*
+ * Constants used in vector implementation of float64 exp(x)
+ */
+#define NPY_RINT_CVT_MAGIC 0x1.8p52
+#define NPY_INV_LN2_MUL_32 0x1.71547652b82fep+5
+#define NPY_TANG_NEG_L1 -0x1.62e42fefp-6
+#define NPY_TANG_NEG_L2 -0x1.473de6af278edp-39
+#define NPY_TANG_A1 0x1p-1
+#define NPY_TANG_A2 0x1.5555555548f7cp-3
+#define NPY_TANG_A3 0x1.5555555545d4ep-5
+#define NPY_TANG_A4 0x1.11115b7aa905ep-7
+#define NPY_TANG_A5 0x1.6c1728d739765p-10
+
+/* Lookup table for 2^(j/32) */
+static npy_uint64 EXP_Table_top[32] = {
+    0x3FF0000000000000,
+    0x3FF059B0D3158540,
+    0x3FF0B5586CF98900,
+    0x3FF11301D0125B40,
+    0x3FF172B83C7D5140,
+    0x3FF1D4873168B980,
+    0x3FF2387A6E756200,
+    0x3FF29E9DF51FDEC0,
+    0x3FF306FE0A31B700,
+    0x3FF371A7373AA9C0,
+    0x3FF3DEA64C123400,
+    0x3FF44E0860618900,
+    0x3FF4BFDAD5362A00,
+    0x3FF5342B569D4F80,
+    0x3FF5AB07DD485400,
+    0x3FF6247EB03A5580,
+    0x3FF6A09E667F3BC0,
+    0x3FF71F75E8EC5F40,
+    0x3FF7A11473EB0180,
+    0x3FF82589994CCE00,
+    0x3FF8ACE5422AA0C0,
+    0x3FF93737B0CDC5C0,
+    0x3FF9C49182A3F080,
+    0x3FFA5503B23E2540,
+    0x3FFAE89F995AD380,
+    0x3FFB7F76F2FB5E40,
+    0x3FFC199BDD855280,
+    0x3FFCB720DCEF9040,
+    0x3FFD5818DCFBA480,
+    0x3FFDFC97337B9B40,
+    0x3FFEA4AFA2A490C0,
+    0x3FFF50765B6E4540,
+};
+
+static npy_uint64 EXP_Table_tail[32] = {
+    0x0000000000000000,
+    0x3D0A1D73E2A475B4,
+    0x3CEEC5317256E308,
+    0x3CF0A4EBBF1AED93,
+    0x3D0D6E6FBE462876,
+    0x3D053C02DC0144C8,
+    0x3D0C3360FD6D8E0B,
+    0x3D009612E8AFAD12,
+    0x3CF52DE8D5A46306,
+    0x3CE54E28AA05E8A9,
+    0x3D011ADA0911F09F,
+    0x3D068189B7A04EF8,
+    0x3D038EA1CBD7F621,
+    0x3CBDF0A83C49D86A,
+    0x3D04AC64980A8C8F,
+    0x3CD2C7C3E81BF4B7,
+    0x3CE921165F626CDD,
+    0x3D09EE91B8797785,
+    0x3CDB5F54408FDB37,
+    0x3CF28ACF88AFAB35,
+    0x3CFB5BA7C55A192D,
+    0x3D027A280E1F92A0,
+    0x3CF01C7C46B071F3,
+    0x3CFC8B424491CAF8,
+    0x3D06AF439A68BB99,
+    0x3CDBAA9EC206AD4F,
+    0x3CFC2220CB12A092,
+    0x3D048A81E5E8F4A5,
+    0x3CDC976816BAD9B8,
+    0x3CFEB968CAC39ED3,
+    0x3CF9858F73A18F5E,
+    0x3C99D3E12DD8A18B,
+};
+
+
```

Comment:
This code does not belong in a public header, and needs to be bracketed with an 
```
if defined HAVE_ATTRIBUTE_TARGET_AVX512F_WITH_INTRINSICS && defined  NPY_HAVE_SSE2_INTRINSICS
```

to avoid compiler warnings about unused variables.

Reply:
Bracket the data still causes unused warning. Move them to separate header file can avoid these warnings. Where should I add the new file?

Reply:
You should create a new header file under `numpy/core/src/umath`. What do you think of the name `numpy/core/src/umath/npy_tables.h`? It would be nice to move the other non-public constants from `npy_math.h` into the new file, I think all the "constansts used in vector ..." can be safely moved.

Reply:
The lookup table and  "constansts used in vector ..." has been moved into `numpy/core/src/umath/npy_simd_data.h`.